ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -112,15 +129,14 @@ targetPeers PeerSelectionActions{readPeerSelectionTargets}
                              LocalRootPeers.keysSet localRootPeers'
 
       return $ \_now -> Decision {
-        decisionTrace = [TraceTargetsChanged targets targets'],
+        decisionTrace = [TraceTargetsChanged targets targets''],
         decisionJobs  = [],
         decisionState = st {
-                          targets        = targets',
+                          targets        = targets'',
                           localRootPeers = localRootPeers',
                           publicRootPeers = publicRootPeers'
-                        }
-      }
-
+                        } }
+    where
```

Comment:
You read a new ledger state judgement value and don't check it against the hold one, so if it is really needed to make sure the value is the most current one you should update it here, however, updating the value has consequences depending on the mode the Governor is in so I think this should be done more carefully. My guess is that you don't really need to read the `lsj` here and can just use the current one

Reply:
I've squashed the no-mutex commit and so this is a non-issue now.